op-node/rollup/derive/channel_bank.go

```diff
@@ -32,24 +32,51 @@ type ChannelBank struct {
 	log log.Logger
 	cfg *rollup.Config
 
-	channels     map[ChannelID]*Channel // channels by ID
-	channelQueue []ChannelID            // channels in FIFO order
+	channels          map[ChannelID]*Channel // channels by ID
+	channelQueue      []ChannelID            // channels in FIFO order
+	processedChannels ProcessedChannels      // processed channels by ID
 
-	prev    NextFrameProvider
-	fetcher L1Fetcher
+	prev NextFrameProvider
+}
+
+// ProcessedChannels tracks channels that have been processed (either fully-read or timed-out).
+// This is to ensure if a new frame arrives for a completed channel, we don't create a new
+// channel for that frame.
+type ProcessedChannels struct {
+	m map[ChannelID]bool
+	a []ChannelID
+}
+
+func NewProcessedChannels() ProcessedChannels {
+	return ProcessedChannels{
+		m: make(map[ChannelID]bool),
+	}
+}
+
+func (p *ProcessedChannels) Seen(id ChannelID) bool {
+	return p.m[id]
+}
+
+func (p *ProcessedChannels) Add(id ChannelID) {
+	p.m[id] = true
+	p.a = append(p.a, id)
+	if len(p.a) > MaxProcessedChannelsSize {
```

Comment:
this could also be time-based if we want to make spec changes to allow re-use of a channelID within a certain time period